src/Transport/Addressing/Partitioning/ReadOnlySettingsExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.AzureServiceBus.Addressing.Partitioning
+{
+    using System.Collections.Generic;
+    using System.Configuration;
+    using System.Linq;
+    using Settings;
+    using Transport.AzureServiceBus;
+
+    public static class ReadOnlySettingsExtensions
+    {
+        public static List<RuntimeNamespaceInfo> GetPartitioningNamespaces(this ReadOnlySettings settings)
```

Comment:
This will internalize `NamespaceConfigurations`, but not `ReadOnlySettings`. 
What the usage code will look like?
```c#
endpointConfiguration.GetSettings().GetPartitioningNamespaces()
```
Then it's counter-intuitive to the extension point `transport.NamespacePartitioning().xyz`